org.eclipse.xtext.testing/src/org/eclipse/xtext/testing/formatter/FormatterTestHelper.java

```diff
@@ -138,18 +139,45 @@ protected void assertNoSyntaxErrors(XtextResource resource) {
 	protected void assertReplacementsAreInRegion(List<ITextReplacement> rep, Collection<ITextRegion> regions,
 			String doc) {
 		Set<ITextReplacement> invalid = Sets.newHashSet();
-		ALLOWED: for (ITextRegion allowed : regions)
-			for (ITextReplacement r : rep) {
-				if (allowed.contains(r))
+		ALLOWED: for (ITextRegion region : regions) {
+			Iterator<ITextReplacement> iterator = rep.iterator();
+			while (iterator.hasNext()) {
+				ITextReplacement replacement = iterator.next();
+				if (region.contains(replacement) || doIntersect(region, replacement))
 					continue ALLOWED;
-				invalid.add(r);
+				
+				if(!iterator.hasNext())
```

Comment:
this if-statement looks bogus to me. Why should only the last replacement be potentially invalid? The list may not be sorted.